### ISSUE TITLE
Use pre-existing library for JWTs?

### ISSUE THREAD

**Issue by JoakimLofgren** [ToastShaman/dropwizard-auth-jwt#21]
It could be nice to use [jose4j](https://bitbucket.org/b_c/jose4j/wiki/Home) or some other library mentioned on the [jwt.io page](http://jwt.io/) ?

**Comment by ToastShaman** [comment]
Hi, that's a very good idea! I've tried jjwt before and quite liked it. Probably worth trying to compare the two to see which one fits better. Have you any experience with either of them?

**Comment by JoakimLofgren** [comment]
I've just used jose4j and can't say I've got any big annoyances with it yet.
I used it for generating JWKs and verifying signatures on JWTs, using RsaJsonWebKey based on the [JWT examples](https://bitbucket.org/b_c/jose4j/wiki/JWT%20Examples).

I like to look at number of commits, releases, open issues and pull requests, activity, and such as well when picking a dependency.


As a side note: If I remember correctly, I noticed that your code breaks on the `jws.setKeyIdHeaderValue(...)` part. So I just omitted that, but it would be nice to use it to identify which key to use when verifying the JWT.
